prov/gni/src/gnix_ep.c

```diff
@@ -576,6 +593,12 @@ int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 		}
 	}
 
-	if (nic != NULL) free(nic);
+	pthread_mutex_lock(&gnix_nic_list_lock);
+	if (incremented_ptag_nic_cnt)
```

Comment:
oh that was to handle cleanup cases by the code below the ```err:``` goto label  I needed to distinguish between when a new nic structure had to be created vs just using a nic that was already in the ```gnix_nic_list``` linked list.  well wait, i see a bug now.  I'll fix it.